### PR TITLE
switch to supervisord

### PR DESCRIPTION
**Pull request by holzgeist**
Similar reasons as in [https://github.com/silvio/matrix-riot-docker/pull/4](https://github.com/silvio/matrix-riot-docker/pull/4), but with multiple processes.

The turnserver is started depending on the presence of /data/turnserver.conf

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,6 +6,11 @@ MAINTAINER Silvio Fricke <silvio.fricke@gmail.com>
 # install homerserver template
 COPY adds/start.sh /start.sh
 
+# add supervisor configs
+COPY adds/matrix.conf /conf/
+COPY adds/turnserver.conf /conf/
```
Comment (silvio) [inline Dockerfile]: Can you reneame this two files? Something like `supervisor-....conf`

File: adds/start.sh [hunk 2]
```diff
@@ -58,7 +58,13 @@ case $OPTION in
...
 		if [ -f /data/turnserver.conf ]; then
 			echo "-=> start turn"
-			/usr/bin/turnserver --daemon -c /data/turnserver.conf
+			if [ -f /conf/turnserver.conf.deactivated ]; then
+				mv -f /conf/turnserver.conf.deactivated /conf/turnserver.conf
+			fi
+		else
+			if [ -f /conf/turnserver.conf ]; then
+				mv -f /conf/turnserver.conf /conf/turnserver.conf.deactivated
+			fi
```
Comment (silvio) [inline adds/start.sh:67]: `turnserver.conf.deactivated`?
You probably mixed the conf files of supervisor and turnserver?

File: adds/supervisord.conf [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[supervisord]
+nodaemon = true
+
+[include]
+files = conf/*.conf
```
Comment (silvio) [inline adds/supervisord.conf]: I don't know supervisior. The messages produced by turnserver and matrix are printed onto console with supervisor?

**Review by silvio — CHANGES_REQUESTED** [review]
My comments and please rebase.

BTW, Thanks for your contribution!

File: adds/start.sh (hunk 2 above)
Comment (xsteadfastx) [reply adds/start.sh:67]: i think its the supervisord config file for turnserver. if there is no `/data/turnserver.conf` then deactivate turnserver in supervisord... and if there is `/data/turnserver.conf` and the turnserver supervisord conf is deactivated... activate it with moving the file.

File: adds/supervisord.conf (hunk 3 above)
Comment (xsteadfastx) [reply adds/supervisord.conf]: i dont know the defaults... but usually you add

```
stdout_logfile=/dev/stdout
stdout_logfile_maxbytes=0

stderr_logfile=/dev/stderr
stderr_logfile_maxbytes=0
```
to `supervisord.conf`.

**Comment by xsteadfastx** [comment]
i really would love to see this merged. i think its a good way to get synapse and turn running. and it would make it easier to run both with a other user than root. supervisord supports `user = nobody` in the program config files.

**Comment by holzgeist** [comment]
Sorry for the late reply. I'm in the end phase of my studies and therefore slightly stressed.

Thanks for the comments, I applied them and pushed.
However, I do not have a lot of experiences with github pull requests, so I don't know if that's enough or there is some button to push?

**Comment by silvio** [comment]
Thanks @holzgeist , I will test you contribution. But this needs time.

**Comment by holzgeist** [comment]
Do not test yet, I tested the actual run too late and found an error in the configuration..

the stdout_logfile et al. should be in [program:x] sections and not in [supervisor]. Rebuilding right now and pushing the result on success

**Comment by xsteadfastx** [comment]
ok. now i was a few minutes too late :) i wanted to see supervisor in the image and did the changes @silvio wanted to see...plus added a user to run the services with and also fixed a bug with logging.

**Comment by holzgeist** [comment]
@silvio: use pull-request #21, I have some troubles getting the log output from both matrix and turnserver to stdout.. 
The extra user is an additional benefit

**Pull request closed without merging**